src/ssd/ssd.c

```diff
@@ -278,13 +304,48 @@ ssd_part_contains(enum ssd_part_type whole, enum ssd_part_type candidate)
 }
 
 void
-ssd_set_active(struct view *view, bool active)
+ssd_set_active(struct ssd *ssd, bool active)
 {
-	if (!view->ssd.tree) {
+	if (!ssd->tree) {
 		return;
 	}
-	wlr_scene_node_set_enabled(&view->ssd.border.active.tree->node, active);
-	wlr_scene_node_set_enabled(&view->ssd.titlebar.active.tree->node, active);
-	wlr_scene_node_set_enabled(&view->ssd.border.inactive.tree->node, !active);
-	wlr_scene_node_set_enabled(&view->ssd.titlebar.inactive.tree->node, !active);
+	wlr_scene_node_set_enabled(&ssd->border.active.tree->node, active);
+	wlr_scene_node_set_enabled(&ssd->titlebar.active.tree->node, active);
+	wlr_scene_node_set_enabled(&ssd->border.inactive.tree->node, !active);
+	wlr_scene_node_set_enabled(&ssd->titlebar.inactive.tree->node, !active);
+}
+
+/* debug */
+struct wlr_scene_node *
+ssd_get_tree_node(const struct ssd *ssd)
+{
+	return ssd->tree ? &ssd->tree->node : NULL;
+}
+
+/* debug */
+const char *
+ssd_get_view_part(const struct ssd *ssd, struct wlr_scene_node *node)
```

Comment:
I am not sure about this, It accesses internal state of the SSD so having it in `src/ssd/` kind of makes sense.
On the other side this is only relevant for the debug action. Maybe we could move these two functions to `src/ssd/ssd_debug.c` and maybe even create a separate `include/ssd_debug.h`? It's really hard for me to figure out the right granularity here.

Also: something like `ssd_debug_name(const struct ssd *ssd, struct wlr_scene_node *node)` would be more self explaining I think and could be expanded in the future so we also have a `view_debug_name()` or similar. So each "module" would describe itself instead of having the external debug.c caller poke around in "module internal" states.

Reply:
Renamed to `ssd_debug_get_node_name()`. Since there are only two `ssd_debug` functions, I don't think we necessarily need to add a new source file + header at this point.

Reply:
Sounds good